edgedb-client/src/builder.rs

```diff
@@ -110,6 +110,29 @@ impl Builder {
                 format!("cannot read credentials file {}", path.display())
         })?)
     }
+    pub fn from_dsn(dsn: &str) -> anyhow::Result<Builder> {
+        if !dsn.starts_with("edgedb://") {
+            anyhow::bail!("String {:?} is not a valid DSN", dsn)
+        };
+        let url = url::Url::parse(dsn)
+            .with_context(|| format!("cannot parse DSN {:?}", dsn))?;
+        Ok(Builder {
+            addr: Addr::Tcp(
+                url.host_str().unwrap_or("127.0.0.1").to_owned(),
```

Comment:
I'd move these, including the port number, to some file with constants and use that across the code base.

Reply:
Well, I don't extract constants unless I plan to change them. This only reduces readability of the code: you need to jump somewhere to find out what the value is. Especially if the value is shorter than descriptive name of the constant :)

Reply:
I'm not a big fan of extracting constants without good reason, but the default port and host are used in multiple places, so I think it makes sense here.

Reply:
Well, while port is used in multiple places in this codebase, it's also used in 5 other repos (and perhaps in some user's repos). So it is close to impossible to change anyway.

That said, for port specifically I would accept PR for making it constant, just for the sake of documentation. It's just I don't care enough and that both ways are equally good for me.